### PR TITLE
Adding Multus and Calico CNI to Minikube

### PR DESCRIPTION
**Pull request by edcdavid**
Adding support for Calico and Multus to Minikube and increasing the default timeout for creating deployments to 240s

### CODE REVIEW

File: scripts/deploy-multus-network.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# Initialization
+SCRIPT_DIR=$(dirname "$0")
+source $SCRIPT_DIR/init-env.sh
+
+rm -rf ./temp
```
Comment (jun-rh) [inline scripts/deploy-multus-network.sh:7]: we don't skip this on ocp?

Comment (edcdavid) [reply scripts/deploy-multus-network.sh:7]: yes correct, will fix

Comment (edcdavid) [reply scripts/deploy-multus-network.sh]: good catch, fixed

File: scripts/deploy-multus-network.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  rm -rf ./temp
+  git clone --depth 1 https://github.com/k8snetworkplumbingwg/multus-cni.git ./temp/multus-cni
+
+  # fix for dimensioning bug
+  sed 's/memory: "50Mi"/memory: "100Mi"/g' temp/multus-cni/deployments/multus-daemonset-thick-plugin.yml -i
+
+  # Deploy Multus
+  cd ./temp/multus-cni
+  cat ./deployments/multus-daemonset-thick-plugin.yml | oc apply -f -
+  cd ../..
```
Comment (hamadise) [inline scripts/deploy-multus-network.sh:22]: I think we can do 
``` oc apply -f ./temp/multus-cni/deployment/multus-daemonset-thick-plugin.yml ```

File: local-test-infra/local-partner-deployment.yaml [hunk 3]
```diff
@@ -16,6 +16,7 @@ spec:
         test-network-function.com/generic: orchestrator
       annotations:
         test-network-function.com/defaultnetworkinterface: '"eth0"' # optional
+        k8s.v1.cni.cncf.io/networks: macvlan-conf # multus network
```
Comment (jun-rh) [inline local-test-infra/local-partner-deployment.yaml:19]: what happens when we deploy this on ocp without the network definition yml?

Comment (edcdavid) [reply local-test-infra/local-partner-deployment.yaml]: fixed, removed the line when deploying in OCP with mustache.

File: scripts/deploy-multus-network.sh (hunk 2 above)
Comment (edcdavid) [reply scripts/deploy-multus-network.sh]: fixed

File: scripts/deploy-multus-network.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+# Initialization
+SCRIPT_DIR=$(dirname "$0")
+source $SCRIPT_DIR/init-env.sh
+
+if $IS_MINIKUBE
```
Comment (jun-rh) [inline scripts/deploy-multus-network.sh:7]: Can we reverse the name and reflect the actual check better? It's actually "NON_OCP". Also, prefixing it with TNF_?

**Review by hamadise — APPROVED** [review]

File: scripts/deploy-multus-network.sh (hunk 4 above)
Comment (edcdavid) [reply scripts/deploy-multus-network.sh]: fixed

**Review by jun-rh — APPROVED** [review]

**Review by hamadise — APPROVED** [review]

File: scripts/deploy-multus-network.sh [hunk 5]
```diff
@@ -16,11 +16,20 @@ then
 
   # Deploy Multus
   oc apply -f ./temp/multus-cni/deployments/multus-daemonset-thick-plugin.yml
+  
+  # Wait for all calico and multus daemonset pods to be running
+  oc rollout status daemonset calico-node -n kube-system  --timeout=240s
```
Comment (jun-rh) [inline scripts/deploy-multus-network.sh:21]: $TNF_DEPLOYMENT_TIMEOUT?

File: README.md [hunk 6]
```diff
@@ -81,18 +81,22 @@ In order to run the local test setup, the following dependencies are needed:
...
 
 ## Setup
+Install the latest minikube by following the instructions at:
+```https://minikube.sigs.k8s.io/docs/start/```
 
 To start minikube, issue the following command:
 
 ```shell-script
-minikube start --embed-certs --driver="virtualbox" --nodes 3
+minikube start --embed-certs --driver="virtualbox" --nodes 3 --network-plugin=cni --cni=calico
```
Comment (jun-rh) [inline README.md:90]: docker?

Comment (javierpena) [reply README.md:90]: Actually, should we specify a driver? Minikube should detect what's available.

**Comment by hamadise** [comment]
tested on minikube in macos. 
networking suite working as expected.

File: README.md (hunk 6 above)
Comment (edcdavid) [reply README.md:89]: I could not get it to work with the virtualbox driver. There is some issue with the dns resolution not working in this configuration. Docker implementation seems ok

File: scripts/deploy-multus-network.sh (hunk 5 above)
Comment (edcdavid) [reply scripts/deploy-multus-network.sh]: fixed

**Review by greyerof — APPROVED** [review]
Tested on minikube v1.22.0 with docker driver in a Fedora 34 vm.
```
Ran 3 of 3 Specs in 36.154 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 0 Skipped
```

**Pull request merged**